stumpy/scrump.py

```diff
@@ -585,13 +684,11 @@ def __init__(
 
         if pre_scrump:
             if self._ignore_trivial:
-                P, I = prescrump(T_A, m, s=s)
+                P, I = prescrump(T_A, m, s=s, k=k)
             else:
-                P, I = prescrump(T_A, m, T_B=T_B, s=s)
-            for i in range(P.shape[0]):
-                if self._P[i, 0] > P[i]:
-                    self._P[i, 0] = P[i]
-                    self._I[i, 0] = I[i]
+                P, I = prescrump(T_A, m, T_B=T_B, s=s, k=k)
```

Comment:
> I say we leave it as the `if/else` because it is inconsistent with other parts of the code that uses `ignore_trivial` to distinguish between self vs AB-joins

Got it :) I will keep the if/else.